### ISSUE TITLE
Sinatra integration not setting rack.request

### ISSUE THREAD

**Issue by jpaulgs** [DataDog/dd-trace-rb#486]
I'm setting up DD APM to trace a Rack -> Sinatra -> Postgres application

With the following configuration each of these layers are presented as a seperate service in the environment.  Which seems to be the way DD is expecting to be setup.

```
  Datadog.configure do |c|
    c.tracer(
      hostname: ENV.fetch('DD_AGENT_SERVICE_HOST'),
      port: ENV.fetch('DD_AGENT_SERVICE_PORT'),
      env: ENV.fetch('DD_APM_ENV')
    )
    c.use :rack
    c.use :sinatra 
    c.use :sequel
  end

  use Datadog::Contrib::Rack::TraceMiddleware
```

![image](https://user-images.githubusercontent.com/30516993/42739233-97546632-88d6-11e8-8e7a-3f32b8f585aa.png)

However the rack service is grouping requests by http method and response code

![image](https://user-images.githubusercontent.com/30516993/42740909-90c20c92-88f0-11e8-881e-6f6c7b0ec59d.png)

Whilst the Sinatra service is working as expected:

![image](https://user-images.githubusercontent.com/30516993/42740920-ab35bf74-88f0-11e8-9356-0f03c4a56812.png)

From reading through the code it seems like the Sinatra integration should be setting the resource on the rack span or am I missing something?

**Comment by delner** [comment]
I do not believe the Sinatra integration is currently setup to do this. Rails does by default, by looking at its parent span, which if its a `'rack.request'`, it then overrides the resource on it. Out of the box, Sinatra won't do this, but it could by retrieving the `'datadog.rack_request_span'` key from the Rack ENV and overriding its resource. (We'd like to find a better way of doing this but that's the current option right now.)

If you don't want to patch the Sinatra integration, you could use the Processing Pipeline to modify the Rack parent span whenever a Sinatra span is processed. Not ideal, but it would probably get things working in the interim.

Let me know if this works for you!

**Comment by jpaulgs** [comment]
I just created a pull request that updates the sinatra middleware to set the rack.resource.

Have a look and see what you think

**Comment by delner** [comment]
@jpaulgs I'll check it out. Thanks!

**Comment by marcotc** [comment]
Thank you for the report again @jpaulgs!

#1015 was just merged, which addresses this issue.
We'll update this GH issue when the release is out with this fix.

**Comment by marcotc** [comment]
[v0.35.0](https://github.com/DataDog/dd-trace-rb/releases/tag/v0.35.0) has been released, with #1015 which address this issue.
But please let us know if there's any further feedback.
